[FionaQY] iP

FionaQY opened the pull request:
### Strand

Strand is Duke, but with it's name changed and **more** features including:
1. *Store* tasks 
 - to store Todos: todo <description> (e.g. `todo homework`)
 - to store Deadlines: deadline /by <deadline> (e.g. `deadline /by tomorrow`)
 - to store Events event /from <start date> /to <end date> (e.g. `event /from 01/01/1010 /to 09/09/1999`)
2. *Save* tasks
3. *List* tasks
4. *Find* tasks

Download latest release from [here](https://github.com/FionaQY/ip/releases).

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

FionaQY committed:
Changed name to Strand

FionaQY committed:
Changed name to Strand

FionaQY committed:
Added Echoing and improved formating

FionaQY committed:
Implement ability to store text entered

FionaQY committed:
Fix minor bugs

FionaQY committed:
Implement ability to mark and unmark tasks

FionaQY committed:
Implement tracking of 3 types of task

FionaQY committed:
Remove unused import

FionaQY committed:
Edit input and expected output files

FionaQY committed:
Implement Strand specific exceptions

FionaQY committed:
Add javadoc comments

FionaQY committed:
Implement deleting of tasks

FionaQY committed:
Refactor code

FionaQY committed:
Implement Enums and update TextUiTesting

FionaQY committed:
Implement saving tasks to hard drive

FionaQY committed:
Add date and time parsing to tasks

FionaQY committed:
Merge Level-8

FionaQY committed:
Merge pull request #1 from FionaQY/branch-Level-8

Add date and time parsing to tasks

FionaQY committed:
Clean up code

FionaQY committed:
Merge remote-tracking branch 'origin/master'

FionaQY committed:
Make code more OOP and divide classes into packages

FionaQY committed:
Merge branch 'branch-MoreOOP'

FionaQY committed:
Merge branch 'master' into add-gradle-support

FionaQY committed:
Implement automation of some build tasks using Gradle

FionaQY committed:
Merge pull request #2 from FionaQY/add-gradle-support

Add gradle support

FionaQY committed:
Add JUnit tests for the Parser and Deadline classes and fix error Task classes

FionaQY committed:
Merge pull request #3 from FionaQY/branch-JUnit

Add JUnit tests for the Parser and Deadline classes and fix error Tas…

FionaQY committed:
Add changes to allow for JAR files

FionaQY committed:
Add on to existing javadoc comments

FionaQY committed:
Merge branch 'branch-A-JavaDoc'

FionaQY committed:
Tweak code to comply with a coding standard

FionaQY committed:
Implement Find Command

FionaQY committed:
Merge branch 'branch-Level-9'

FionaQY committed:
Fix coding style violations

FionaQY committed:
Merge branch 'branch-A-CheckStyle'

FionaQY committed:
Merge pull request #4 from FionaQY/branch-A-CheckStyle

Fix coding style violations

FionaQY committed:
Merge branch 'master' of github.com:FionaQY/ip

FionaQY committed:
Implement GUI to Strand

FionaQY committed:
Merge branch 'branch-Level-10'

FionaQY committed:
Merge pull request #5 from FionaQY/branch-Level-10

Implement GUI for Strand

FionaQY committed:
Merge branch 'master' of github.com:FionaQY/ip

WeiHanWong commented on src/main/java/MainWindow.java:
Could put this under a package for UIs

WeiHanWong commented on src/main/java/DialogBox.java:
Could put this under a package for UIs

WeiHanWong reviewed (commented):
LGTM. Just some nits to fix.

kellywsq03 commented on src/main/java/strand/Parser.java:
```diff
@@ -0,0 +1,173 @@
+        CommandEnums command = getCommand(split[0].toUpperCase());
+
+        switch (command) {
+        case DELETE, MARK, UNMARK: {
```
Great use of Enums! Good job

kellywsq03 replied:
Great use of Enums! Good job

kellywsq03 commented on src/main/java/strand/Storage.java:
```diff
@@ -0,0 +1,116 @@
+                newTask.markAsDone();
+            }
+        }
+        case "D" -> newTask = new Deadline(split[2], split[3]);
```
Perhaps you may want to follow the coding conventions of this course

kellywsq03 commented on src/main/java/strand/Ui.java:
```diff
@@ -0,0 +1,116 @@
+    /**
+     * Displays a welcome message to the user.
+     */
+    public String welcome() {
```
Cute interactions with user. Would make for a great UX

kellywsq03 reviewed (commented):
LGTM!

FionaQY committed:
Add Assertions, Update Javadoc Comments, and Rename Methods Across Multiple Classes

- Introduced assertions in relevant methods to ensure non-null values and validate conditions.
- Updated Javadoc comments across classes for improved clarity and completeness.
- Renamed methods in `TaskList`, `Storage`, `Parser`, and other affected classes for better readability and consistency.

Changes:
- **TaskList**:
  - Added assertions in constructor to check for non-null task lists.

- **Storage**:
  - Added assertions to verify file path and ensure file operations are performed correctly.

- **Parser**:
  - Implemented assertions to validate user inputs and command parsing logic.

- **Task Classes**:
  - Renamed methods for better readability and adherence to conventions.

FionaQY committed:
Improve Code Quality

Refactored code to enhance readability and maintainability. Updates include:

    Improved method names and error handling in the Strand class.
    Optimized stream operations and method implementations in the TaskList class.

FionaQY committed:
Merge branch 'branch-A-Assertions'

FionaQY committed:
Merge pull request #6 from FionaQY/branch-A-Assertions

Add Assertions, Update Javadoc Comments, and Rename Methods

FionaQY committed:
Merge branch 'master' of github.com:FionaQY/ip

FionaQY committed:
Resolve merge conflicts

FionaQY committed:
Merge pull request #7 from FionaQY/branch-A-CodeQuality

Improve Code Quality

FionaQY committed:
Implement extension C-Priority and update ParserTest for this extension

FionaQY committed:
Merge pull request #8 from FionaQY/branch-BCD-Extension

Implement extension C-Priority and update ParserTest

FionaQY committed:
Change implementation of dates for tasks

FionaQY committed:
Fix javadoc comment

FionaQY committed:
Fix boolean name

FionaQY committed:
Add screenshot and set up User Guide

FionaQY committed:
Update README.md

FionaQY committed:
Fix bug where jar created doesn't have gui

FionaQY committed:
Merge branch 'master' of github.com:FionaQY/ip

FionaQY committed:
Update README.md

FionaQY committed:
Update README.md

FionaQY committed:
Update README.md

FionaQY committed:
Update README.md

FionaQY committed:
Update README.md

FionaQY committed:
Update README.md

FionaQY committed:
Update README.md

FionaQY committed:
Update javadoc comments

FionaQY committed:
Add more tests

FionaQY committed:
Fix README.md
